lib/LoaderOptionsPlugin.js

```diff
@@ -16,17 +16,23 @@ class LoaderOptionsPlugin {
 	}
 
 	apply(compiler) {
-		let options = this.options;
+		const options = this.options;
+		const filterSet = new Set(["include", "exclude", "test"]);
 		compiler.plugin("compilation", (compilation) => {
 			compilation.plugin("normal-module-loader", (context, module) => {
-				let resource = module.resource;
+				const resource = module.resource;
 				if(!resource) return;
-				let i = resource.indexOf("?");
+				const i = resource.indexOf("?");
 				if(ModuleFilenameHelpers.matchObject(options, i < 0 ? resource : resource.substr(0, i))) {
-					let filterSet = new Set(["include", "exclude", "test"]);
 					Object.keys(options)
 						.filter((key) => !filterSet.has(key))
-						.forEach((key) => context[key] = options[key]);
+						.forEach((key) => {
+							if(context[key]) {
+								Object.assign(context[key], options[key]);
```

Comment:
Doesn't this modify the original value?

Maybe `context[key] = Object.assign({}, context[key], options[key])` is the better way.

Also make sure to only do this for `objects`